access-token-management/src/AccessTokenManagement/ClientCredentialsTokenManagementOptions.cs

```diff
@@ -8,13 +8,41 @@ namespace Duende.AccessTokenManagement;
 /// </summary>
 public class ClientCredentialsTokenManagementOptions
 {
+    public ClientCredentialsTokenManagementOptions()
+    {
+        GenerateCacheKey = ((clientName, parameters) =>
+        {
+            var s = "s_" + parameters?.Scope ?? "";
+            var r = "r_" + parameters?.Resource ?? "";
+
+            return CacheKeyPrefix + clientName + "::" + s + "::" + r;
+        });
+
+        GenerateNonceStoreKey = (context => $"{GenerateNonceStoreKey}:{context.Url}:{context.Method}");
+    }
+
     /// <summary>
     /// Used to prefix the cache key
     /// </summary>
     public string CacheKeyPrefix { get; set; } = "Duende.AccessTokenManagement.Cache::";
 
+    public string NonceStoreKeyPrefix { get; set; } = "Duende.AccessTokenManagement.DPoPNonceStore::";
+
     /// <summary>
     /// Value to subtract from token lifetime for the cache entry lifetime (defaults to 60 seconds)
     /// </summary>
     public int CacheLifetimeBuffer { get; set; } = 60;
+
+    /// <summary>
+    /// The logic to generate a cache key. Defaults to a simple key based on client name, scope, and resource.
```

Comment:
"based on CacheKeyPrefix, client name...". Perhaps remove "simple".

Reply:
updated the implementation to better resolve the implementation